hash.c

```diff
@@ -1642,6 +1667,24 @@ static void
 rb_hash_modify_check(VALUE hash)
 {
     rb_check_frozen(hash);
+
+    if (hash_copy_on_write_p(hash)) {
+        if (RHASH_AR_TABLE_P(hash)) {
+            // struct ar_table *table = RHASH_AR_TABLE(hash);
+            // if (table) {
+            //     table = ar_copy(table);
+            //     RHASH_AR_TABLE_SET(hash, table);
+            // }
+            rb_bug("Copy on write set for ar_table!");
+        }
+        else if (RHASH_ST_TABLE_P(hash)) {
+            struct st_table *table = RHASH_ST_TABLE(hash);
+            if (table) {
+                RHASH_ST_TABLE_SET(hash, st_copy(table));
+            }
+            hash_copy_on_write_clear(hash);
+        }
+    }
```

Comment:
I'm not a fan of a function that is supposed to only perform checks has the side-effect of potentially modifying the hash. It would make more sense to me if this function was renamed to `rb_hash_modify` (i.e. drop the `check` from the name). It also makes the name consistent with the `rb_ary_modify` function.